vlapy/manager.py

```diff
@@ -118,97 +116,92 @@ def start_run(all_params, pulse_dictionary, diagnostics, name="test"):
+            xmax = all_params["xmax"]
+            xmin = all_params["xmin"]
+            dx = (xmax - xmin) / nx
+            x = np.linspace(xmin + dx / 2.0, xmax - dx / 2.0, nx)
+            kx = np.fft.fftfreq(x.size, d=dx) * 2.0 * np.pi
+
+            # Velocity grid
+            vmax = all_params["vmax"]
+            dv = 2 * vmax / nv
+            v = np.linspace(-vmax + dv / 2.0, vmax - dv / 2.0, nv)
+            kv = np.fft.fftfreq(v.size, d=dv) * 2.0 * np.pi
+
+            t = np.linspace(0, tmax, nt)
+            dt = t[1] - t[0]
+
+            def driver_function(x, tt):
+                total_field = np.zeros_like(x)
+
+                for this_pulse in list(pulse_dictionary.keys()):
+                    kk = pulse_dictionary[this_pulse]["k0"]
+                    ww = pulse_dictionary[this_pulse]["w0"]
+
+                    envelope = get_pulse_coefficient(
+                        pulse_profile_dictionary=pulse_dictionary[this_pulse], tt=tt
+                    )
+
+                    if np.abs(envelope) > 0.0:
+                        total_field += envelope * np.cos(kk * x - ww * tt)
+
+                return total_field
+
+            e = field.get_total_electric_field(
+                driver_function(x=x, tt=t[0]), f=f, dv=dv, kx=kx
             )
 
-            if nu > 0.0:
-                f = lenard_bernstein.take_collision_step(leftside=leftside, f=f,)
+            # Storage
+            storage_manager = storage.StorageManager(
+                x, v, t, temp_path, store_f=diagnostics.f_rules
+            )
+            storage_manager.write_parameters_to_file(all_params, "all_parameters")
+            storage_manager.write_parameters_to_file(pulse_dictionary, "pulses")
 
-            # All storage stuff here
-            storage_manager.temp_update(
-                tt=t[it], f=f, e=e, driver=driver_function(x=x, tt=t[it])
+            # Matrix representing collision operator
+            leftside = lenard_bernstein.make_philharmonic_matrix(
+                v=v, nv=nv, nu=nu, dt=dt, dv=dv, v0=1.0
             )
 
-        # Diagnostics
-        diagnostics(storage_manager)
+            # Time Loop
+            for it in tqdm(range(nt)):
+                e, f = step.full_PEFRL_ps_step(
+                    f, x, kx, v, kv, dv, t[it], dt, e, driver_function
+                )
+
+                if nu > 0.0:
+                    f = lenard_bernstein.take_collision_step(leftside=leftside, f=f,)
+
+                # All storage stuff here
+                storage_manager.temp_update(
+                    current_time=t[it], f=f, e=e, driver=driver_function(x=x, tt=t[it])
+                )
 
-        # Log
-        storage_manager.close()
-        mlflow.log_artifacts(temp_path)
+            # Diagnostics
+            diagnostics(storage_manager)
 
-        # Cleanup
-        shutil.rmtree(temp_path)
+            # Log
+            storage_manager.close()
+            mlflow.log_artifacts(temp_path)
```

Comment:
Here's the line that's concerning!

Reply:
Yeah, this command actually moves the stuff in `temp_path` to a more permanent store.

In the future, one could imagine having an MLFlow server always running on the cloud, and slinging artifacts to Cloud Storage like GCP, S3, and so on, while storing parameters in a DB locally.

Then the user would always run VlaPy locally (or wherever!), connect to the MLFlow server of your choice before your run starts, report the results back to that MLFlow server on the cloud, and disengage. It's a way of doing distributed, and parallel, simulations and keeping track of them in one place!